cpp/src/reductions/compound.cuh

```diff
@@ -53,35 +58,39 @@ std::unique_ptr<scalar> compound_reduction(column_view const& col,
 {
   auto const valid_count = col.size() - col.null_count();
 
+  // All null input produces all null output
+  if (valid_count == 0 ||
+      // Only care about ddof for standard deviation and variance right now
+      valid_count <= ddof && (std::is_same_v<Op, cudf::reduction::detail::op::standard_deviation> ||
+                              std::is_same_v<Op, cudf::reduction::detail::op::variance>)) {
```

Comment:
@bdice Done. I reordered the condition purely because clang-format does a bad for:

```
(std::is_same_v<...> || std::is_same_v<...>) && ...
```
it sticks the `&&` to the paren.

Reply:
`std::is_same_v` is `constexpr`. So you maybe this is better:
```
if constexpr(std::is_same_v<...> || std::is_same_v<...>) {
  if(valid_count == 0 || ...) {
    ...
  }
}
```

Reply:
I am perhaps naive in that I hope that any decent compiler will optimise `if (x && true)` to `if (x)`.

Reply:
Ah, that's not the right transformation, because the `valid_count == 0` branch should be taken for any `Op`.

Reply:
I've found the compiler is pretty good at this sort of thing without our help anyway.